transmart-rest-api-e2e/src/test/groovy/tests/rest/access/AccessCheckSpec.groovy

```diff
@@ -0,0 +1,97 @@
+     */
+    def "access to resources is blocked with an invalid token"() {
+        when: "I try to fetch all studies without access token"
+        def responseData = get([
+                path      : PATH_STUDIES,
+                acceptType: JSON,
+                user      : UNRESTRICTED_USER,
+                statusCode: 401
+        ])
+        then: "I am not authorized to access the resource"
+        responseData.message == "Unauthorized"
+
+        when: "I try to fetch all studies with invalid access token"
+        def responseDataInvalidToken = responseData
+        responseDataInvalidToken.token = 'invalidToken'.bytes.encodeBase64().toString()
+        then: "I am not authorized to access the resource"
+        responseDataInvalidToken.message == "Unauthorized"
+    }
+
+    /**
+     * when: "I try to fetch all studies with a proper access token"
+     * then: "I have an access to the resource"
+     *
+     * when: "I fabricate the token by replacing original user data with a new user"
+     * then: "The signature is verified as invalid, I am not authorized to access the resource"
+     */
+    def "access token validation restricted per user"() {
+
+        when: "I try to fetch all studies with a proper access token"
+        def tokenForUnrestrictedUser = OauthAdapter.getToken(UNRESTRICTED_USER)
+
+        then: "I have an access to the resource"
+        get([
+                path      : PATH_STUDIES,
+                acceptType: JSON,
+                statusCode: 200,
+                token     : tokenForUnrestrictedUser
+        ])
+
+
+        when: "I fabricate the token by replacing original user data with a new user"
+        def tokenWithReplacedUser = replaceUserInToken(tokenForUnrestrictedUser, DEFAULT_USER)
+        def responseForNewUser = get([
+                path      : PATH_STUDIES,
+                acceptType: JSON,
+                token     : tokenWithReplacedUser,
+                statusCode: 401
+        ])
+
+        then: "The signature is verified as invalid, I am not authorized to access the resource"
+        responseForNewUser.error == "Unauthorized"
+        responseForNewUser.message == "Unable to authenticate using the Authorization header"
+    }
+
+    private String replaceUserInToken(String tokenForUnrestrictedUser, String newUser) {
+        // Token is represented as base64 encoded: <HEADER>.<PAYLOAD>.<SIGNATURE>
+        def tokenParts = tokenForUnrestrictedUser.split("\\.")
+        Map tokenPayload = encodeBase64ToMap(tokenParts)
+
+        tokenPayload.given_name = newUser
+        tokenPayload.name = newUser
+        tokenPayload.preferred_username = newUser
+        tokenPayload.sub = getUsername(newUser)
+
+        String newTokenPayload = decodeMapToBase64String(tokenPayload)
+        tokenParts[1] = newTokenPayload
+
+        tokenParts.join(".")
+    }
+
+    private String decodeMapToBase64String(Map tokenPart) {
+        def tokenPartString = toJSON(tokenPart)
```

Comment:
In the other PR I removed the `toJSON` function. Could you use `new ObjectMapper().writeValueAsString(tokenPart)` instead? That would reduce the merge conflicts.